triton-client/src/store/thunks.ts

```diff
@@ -31,50 +33,40 @@ export const fetchProjects = () => async (dispatch: AppDispatch, getState: () =>
 		const projects = await apiTriton.listProjects()
 		// console.debug(`Loaded projects succesfully: ${projects}`)
 		dispatch(ProjectsStateActions.setProjects(projects))
-		dispatch(RunsStateActions.initializeRunsByProjectId(projects))
-		dispatch(DatasetsStateActions.initializeDatasetsByProjectId(projects))
 	} catch (err: any) {
 		dispatch(ProjectsStateActions.setError(convertToSerializedError(err)))
 		throw err
 	}
 }
 
 export const fetchRuns = (externalProjectId: ExternalProjectID) => async (dispatch: AppDispatch, getState: () => RootState) => {
-	if (getState().runsState.runsByProjectId[externalProjectId]?.loading) return
-
 	try {
-		dispatch(RunsStateActions.setLoading(externalProjectId))
 		const runs = await apiTriton.listRunsForProjects([externalProjectId])
 		// console.info('Loaded runs succesfully', runs)
-		dispatch(RunsStateActions.setRunsByProjectId({ projectId: externalProjectId, runs }))
+		dispatch(RunsStateActions.setRuns(runs))
 		// console.info(getState().runsState.runsById)
 		return runs
 	} catch (err: any) {
-		dispatch(DatasetsStateActions.setError({ projectId: externalProjectId, error: convertToSerializedError(err) }))
 		throw err
 	}
 }
 
-export const fetchDatasets = (externalProjectId: ExternalProjectID) => async (dispatch: AppDispatch, getState: () => RootState) => {
-	if (getState().datasetsState.datasetsByProjectId[externalProjectId]?.loading) return
-	
+export const fetchDatasets = (runName: TritonRun['name']) => async (dispatch: AppDispatch, getState: () => RootState) => {
 	try {
-		dispatch(DatasetsStateActions.setLoading(externalProjectId))
-		const datasetIds = Object.values(getState().runsState.runsByName).reduce<number[]>((datasetIDs, run) => {
-			if (run) {
-				datasetIDs.push(...run.datasets)
-			}
-			return datasetIDs
-		}, [])
+		const run = getState().runsState.runsByName[runName]
+		if (!run) {
+			return []
+		}
+
 		// console.info(datasetIds)
-		const datasets = await apiTriton.listDatasetsByIds(datasetIds)
+		const datasets = await apiTriton.listDatasetsByIds(run.datasets)
 		// console.debug(`Loaded datasets succesfully: ${JSON.stringify(datasets)}`)
-		dispatch(DatasetsStateActions.setDatasetsByProjectId({ projectId: externalProjectId, datasets }))
-		dispatch(ReadsetsStateActions.initializeReadsetsByDatasetIds(datasets))
+		dispatch(DatasetsStateActions.setDatasets(datasets))
+
+		datasets.forEach((dataset) => dispatch(RequestsStateActions.setRequests(dataset.requests)))
```

Comment:
Mmm this line is pretty suspicious ... why are requests coming from dataset ? If you really want to make request independant move the request listing out of listDatasetsByIds ...

Reply:
Yeah. I agree with that too.

Reply:
Another option is to have a single request attached optionally to dataset.